homeassistant/components/plugwise/select.py

```diff
@@ -1,64 +1,115 @@
 """Plugwise Select component for Home Assistant."""
 from __future__ import annotations
 
-from homeassistant.components.select import SelectEntity
+from collections.abc import Awaitable, Callable
+from dataclasses import dataclass
+from typing import Any
+
+from homeassistant.components.select import SelectEntity, SelectEntityDescription
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import STATE_ON
 from homeassistant.core import HomeAssistant
-from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity import EntityCategory
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
-from .const import DOMAIN, THERMOSTAT_CLASSES
+from .const import DOMAIN
 from .coordinator import PlugwiseDataUpdateCoordinator
 from .entity import PlugwiseEntity
 
 
+@dataclass
+class PlugwiseSelectDescriptionMixin:
+    """Mixin values for Plugwise Select entities."""
+
+    command: Callable[..., Awaitable[Any]]
+    current_option: str
+    options: str
+
+
+@dataclass
+class PlugwiseSelectEntityDescription(
+    SelectEntityDescription, PlugwiseSelectDescriptionMixin
+):
+    """Class describing Plugwise Number entities."""
+
+
+SELECT_TYPES = (
+    PlugwiseSelectEntityDescription(
+        key="select_schedule",
+        name="Thermostat Schedule",
+        icon="mdi:calendar-clock",
+        command=lambda coordinator, location, option: coordinator.api.set_schedule_state(
+            location, option, STATE_ON
+        ),
```

Comment:
We generally don't want lambda to cover multiple lines...
So this works around that a little 😉 

```suggestion
        command=lambda api, loc, opt: api.set_schedule_state(loc, opt, STATE_ON),
```